linebot/api.py

```diff
@@ -1148,6 +1149,244 @@ def get_bot_info(self, timeout=None):
+
+        :param str audience_group_id: The audience ID
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        """
+        self._delete(
+            '/v2/bot/audienceGroup/{}'.format(audience_group_id),
+            timeout=timeout
+        )
+
+    def rename_audience_group(self, audience_group_id, audience_group_name, timeout=None):
+        """Modify the name of an existing audience.
+
+        https://developers.line.biz/en/reference/messaging-api/#set-description-audience-group
+
+        :param str audience_group_id: The audience ID
+        :param str audience_group_name: The new audience's name
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        """
+        self._put(
+            '/v2/bot/audienceGroup/{}/updateDescription'.format(audience_group_id),
+            data=json.dumps({
+                "description": audience_group_name,
+            }),
+            timeout=timeout
+        )
+
+    def add_audiences_to_audience_group(self, audience_group_id, audiences, is_ifa=False,
+                                        upload_description=None, timeout=None):
+        """Add new user IDs or IFAs to an audience for uploading user IDs.
+
+        https://developers.line.biz/en/reference/messaging-api/#update-upload-audience-group
+
+        :param str audience_group_id: The audience ID
+        :param list audiences: An array of user IDs or IFAs
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :param bool is_ifa: If this is false (default), recipients are specified by user IDs.
+            If true, recipients must be specified by IFAs.
+        :param str upload_description: The description to register for the job
+        :type timeout: float | tuple(float, float)
+        """
+        audiences_chunks = [audiences[i:i + 10000] for i in range(0, len(audiences), 10000)]
+        for _audiences in audiences_chunks:
+            self._put(
+                '/v2/bot/audienceGroup/upload',
+                data=json.dumps({
+                    "audienceGroupId": audience_group_id,
+                    "audiences": [{"id": audience} for audience in _audiences],
+                    "isIfaAudience": is_ifa,
+                    "uploadDescription": upload_description,
+                }),
+                timeout=timeout
+            )
+
```

Comment:
We need to return `{}`

refer to https://developers.line.biz/en/reference/messaging-api/#update-upload-audience-group-response

Reply:
modified